src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt

```diff
@@ -136,6 +142,26 @@ class BookingExtractionService(
     )
   }
 
+  // Changed logic for Home Detention Curfew Eligibility Date 365 (HDC-365)
+  // Eventually only HDC365 will apply (after commencement date)
+  private fun setHdcedDate(
+    sentenceCalculation: SentenceCalculation,
+    sentence: CalculableSentence,
+    dates: MutableMap<ReleaseDateType, LocalDate>,
+  ) {
+    if (sentenceCalculation.homeDetentionCurfewEligibilityDate != null && !sentence.releaseDateTypes.contains(PED)) {
```

Comment:
lots of chained ifs:

```private fun setHdcedDate(
    sentenceCalculation: SentenceCalculation,
    sentence: CalculableSentence,
    dates: MutableMap<ReleaseDateType, LocalDate>,
) {
    val hdced = sentenceCalculation.homeDetentionCurfewEligibilityDate
    val hdced365 = sentenceCalculation.homeDetentionCurfewEligibilityDateHDC365
    val commencementDate = ImportantDates.HDC_365_COMMENCEMENT_DATE

    if (hdced != null && !sentence.releaseDateTypes.contains(PED)) {
        if (hdcedExtractionService.releaseDateIsAfterHdced(sentenceCalculation)) {
            dates[HDCED] = when {
                hdced.isBefore(commencementDate) -> hdced
                hdced365 != null && hdced365.isBefore(commencementDate) -> commencementDate
                else -> hdced365 ?: commencementDate // Fallback in case hdced365 is null
            }
        }
    }
}```